File: eval/code_corpus/vulnerable/xss.js
// # INTENTIONALLY VULNERABLE — eval fixture, do not deploy
// DOM XSS — untrusted value written to innerHTML.
// Ground-truth finding: vuln_class=xss on the innerHTML line below.

function showName(name) {
  const el = document.getElementById("greeting");
  // VULN: innerHTML with attacker-controlled string parses & runs markup — XSS sink.
  el.innerHTML = "Hello " + name;
}

module.exports = { showName };

File: eval/code_corpus/vulnerable/xss.py
# INTENTIONALLY VULNERABLE — eval fixture, do not deploy
"""Reflected XSS — user input echoed into an HTML response without escaping.

Ground-truth finding: vuln_class=xss on the return line below.
"""
from flask import request


def greet():
    name = request.args.get("name", "")
    # VULN: raw f-string into HTML body — reflected XSS sink.
    return "<h1>Hello " + name + "</h1>"
